packages/core/core/src/factories/apiFactory/types.ts

```diff
@@ -0,0 +1,51 @@
+import { IntegrationPlugin } from './../../utils/nuxt';
+export interface ApiExtensionInstance {
+  beforeSetup?: (config, headers?: Record<string, string>) => any;
```

Comment:
why not object

Reply:
extension always is depending on request so it's, more clean to have function that takes request and now you have access to it across the lifecycle functions instead of adding extra argument for each:

```js
const extension = (req) => ({
 beforeSetup: (config) => { have access to req }
 beforeCall: (params) => { have access to req }
 ...
})
```

instead of:


```js
const extension = {
 beforeSetup: (req, config) => { extra argument needed }
 beforeCall: (req, params) => {extra argument needed}
}
```